R/factset.protobuf.stachextensions/R/V2RowOrganizedStachExtension.R

```diff
@@ -195,13 +195,31 @@ V2RowOrganizedStachExtension <- R6::R6Class(
         }
         else
         {
-          routeMap <- tableData$data$rows[[rowindex]]$values$fields
-          for (j in 1:defColCount)
-          {
-            columnValue <- routeMap[[j]]$value
+          rowDataMap <- tableData$data$rows[[rowindex]]$values$fields
+
+          # here fields are of type list.
+          # In order to read the value using columnid from fields,
+          # keys and values of fields are converted to key-value pair using data frame.
+
+          fieldsKeys <- c() # stores the fields keys (example:"col_0")
+          fieldsValues <- c() # stores the fields values (example:"Utilities")
+
+          for(v in 1:length(rowDataMap)){
```

Comment:
very minor: please use relevant names.

Reply:
As discussed in offline, R stach schema is defined in such a way that field values are of type list, here as in other languages field values are not of dictionary type, if it is dictionary type there will be a possibility to read the field value directly using key can avoid for loop. So here in the case of R to make the field value type dictionary explicitly converted all the field values to data frame and made it dictionary type.

Also I have tested the snippet (with for loop in stach extension code) with large post body (which returns more than 1600 rows) took 4 minutes, and 11 seconds where as I have made a temporary fix (by reading direct string and sanitizing the string) to avoid for loop which took almost 4 minutes, and 18 seconds. so with out for loop also there is no much difference.

Reply:
I have ran the code snippet **without bug fix** and **with bug fix** to evaluate the performance, on an average the new fix has taken more 19 seconds to complete to process the stach response when compared with code without bug fix.